### PR TITLE
Adding Unity support for Airsim. Windows only for now.

### PR DESCRIPTION
**Pull request by kenbier**
## TODO
- [x] Add build docs
- [x] UnityDistanceSensor::getRayLength() - Need this info from Unity using pinvoke
- [x] CarPawnApi::getCarState() - GetEngineRotationSpeed, GetEngineMaxRotationSpeed need to be implemented
- [x] CarPawnApi::reset() - and few more places need to evaluate option of using threads for better performance.
- [x] Remove any dead code
- [x]  Replace own `Logger.cpp` with `apache's Log4Cxx`
- [x] Add build.cmd script for building AirsimWrapper
- [x] Find a drone with reasonable size and MIT or BSD license approved
- [ ] Convert_to_AirsimResponse() - Need to get camera_name from source. Currently it crashes when there is a string in structure used for data transfer in PInvoke.

### CODE REVIEW

**Comment by ankitdubeyryt** [comment]
> 
> ## TODO
> 
>     * [x]  Replace zipped drone with compressed version (currently over 100 MB)
> 
>     * [ ]  Add build docs
> 
>     * [ ]  Add to build.cmd to build AirLibWrapper
> 
>     * [ ]  UnityDistanceSensor::getRayLength() - Need this info from Unity using pinvoke
> 
>     * [X]  CarPawnApi::getCarState() - GetEngineRotationSpeed, GetEngineMaxRotationSpeed need to be implemented
> 
>     * [X]  CarPawnApi::reset() - and few more places need to evaluate option of using threads for better performance.
> 
>     * [ ]  Convert_to_AirsimResponse() - Need to get camera_name from source. Currently it crashes when there is a string in structure used for data transfer in PInvoke.
> 
>     * [ ]   Replace own `Logger.cpp` with `apache's Log4Cxx`
> 
>     * [X]  Remove any dead code
>     * [ ] Add check for missing `Settings.json` file.

**Review by sytelus — CHANGES_REQUESTED** [review]
These changes are not needed?

File: AirLib/AirLib.vcxproj [hunk 1]
```diff
@@ -286,7 +286,7 @@
       <WarningLevel>Level4</WarningLevel>
       <Optimization>Disabled</Optimization>
       <PreprocessorDefinitions>_SCL_SECURE_NO_WARNINGS;_CRT_SECURE_NO_WARNINGS;_DEBUG;_LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (sytelus) [inline AirLib/AirLib.vcxproj]: Changes in this file doesn't seem to be needed

File: AirLib/include/vehicles/multirotor/firmwares/simple_flight/firmware/Params.hpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ struct Params {
 
     struct VelocityPid {
         const float kMinThrottle = std::min(1.0f, Params::min_armed_throttle() * 3.0f);
-        const float kMaxLimit = 6.0f; // m/s
```
Comment (sytelus) [inline AirLib/include/vehicles/multirotor/firmwares/simple_flight/firmware/Params.hpp]: Please remove unnecessary changes in this file.

File: setup.sh [hunk 3]
```diff
@@ -31,8 +31,8 @@ if [ "$(uname)" == "Darwin" ]; then # osx
     brew tap llvm-hs/homebrew-llvm
     brew install llvm-5.0
 
-    brew install wget
-    brew install coreutils
```
Comment (sytelus) [inline setup.sh]: Could you please explain why this is commented?

File: Unity/DroneDemo/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# Unity AirSim Drone Demo
+
```
Comment (sytelus) [inline Unity/DroneDemo/README.md]: Could have Build section that specifies steps for creating the build? We should start with where and which version of Unity to install.

File: Unity/DroneDemo/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          "AllowAPIAlways": true
+        },
+        "ApiServerPort": 41451
+      }
+>>>
+
+### Keyboard Controls
+
+>>>
+    WASD\Arrow keys: Drone movement.
```
Comment (sytelus) [inline Unity/DroneDemo/README.md]: If you are using AirLib then drone can only be flown using RC. Is there any additional code in Unity port to enable these keys?

**Review by sytelus — CHANGES_REQUESTED** [review]
The drone.obj is very large file in the repo. Can we reduce its size as we discussed? If you have high resolution assets, we can add that separately as zip file in Releases section.

If you can update the README with build and usage instructions, I can start testing on my machine.

File: setup.sh (hunk 3 above)
Comment (kenbier) [reply setup.sh]: Mistaken commit. I will remove

**Comment by kenbier** [comment]
@shital I removed the drone file and undid the changes to the AirLib directory. 
Docs coming next

**Comment by kenbier** [comment]
docs are added. I will wait on squashing until the TODO's are finished.

File: Unity/DroneDemo/README.md (hunk 4 above)
Comment (kenbier) [reply Unity/DroneDemo/README.md]: done

**Comment by kenbier** [comment]
@sytelus, won't you mind reverting the merge of this branch into master? We have done a rebase to get rid of large objects from the git history

**Pull request merged**